src/chronify/time_series_mapper_datetime.py

```diff
@@ -40,79 +54,160 @@ def __init__(
 
-    def _check_time_length(self) -> None:
+    def _check_time_resolution_and_length(self) -> None:
+        """Check that from_schema time resolution and length can produce to_schema counterparts."""
         flen, tlen = self._from_time_config.length, self._to_time_config.length
-        if flen != tlen and not self._wrap_time_allowed:
-            msg = f"Length must match between {self._from_schema.__class__} from_schema and {self._to_schema.__class__} to_schema. {flen} vs. {tlen} OR wrap_time_allowed must be set to True"
+        fres, tres = self._from_time_config.resolution, self._to_time_config.resolution
+        if fres == tres and flen != tlen and not self._wrap_time_allowed:
+            msg = f"For unchanging time resolution, length must match between from_time_config and to_time_config. {flen} vs. {tlen} OR wrap_time_allowed must be set to True"
+            raise ConflictingInputsError(msg)
+
+        # Resolution must be mutiples of each other
+        if fres < tres and tres.total_seconds() % fres.total_seconds() > 0:
+            msg = f"For aggregation, the time resolution in to_time_config must be a multiple of that in from_time_config. {tres} vs {fres}"
+        if fres > tres and fres.total_seconds() % tres.total_seconds() > 0:
+            msg = f"For disaggregation, the time resolution in from_time_config must be a multiple of that in to_time_config. {fres} vs {tres}"
+
+        # No extrapolation allowed
+        if flen * fres.total_seconds() < tlen * tres.total_seconds():
+            msg = "The product of time resolution and length in from_time_config cannot be greater than that in to_time_config."
             raise ConflictingInputsError(msg)
 
+    def _check_resampling_consistency(self) -> None:
+        """Check resampling operation type is consistent with time resolution inputs."""
+        if self._resampling_type is None and self._resampling_operation is not None:
+            msg = f"For unchanging time resolution, {self._resampling_operation=} must be set to None."
+            raise ConflictingInputsError(msg)
+
+        for typ, opt in [("aggregation", AggregationType), ("disaggregation", DisaggregationType)]:
+            if self._resampling_type == typ and not isinstance(self._resampling_operation, opt):
+                msg = f"{typ} detected, {self._resampling_operation=} must be set to an option from {opt}"
+                raise ConflictingInputsError(msg)
+
+    def _create_intermediate_schema(self) -> TableSchema:
+        """Create the intermediate table schema for all time processing except resampling,
+        by producing a version of to_schema with the same time resolution and length as from_schema
+        """
+        schema_kwargs = self._to_schema.model_dump()
+        schema_kwargs["time_config"]["resolution"] = self._from_time_config.resolution
+        schema_kwargs["time_config"]["length"] = self._from_time_config.length
+        schema_kwargs["name"] += "_intermediate"
```

Comment:
User report: this intermediate table isn't getting deleted.

Reply:
Ah, I just fixed those while debugging for Tengis.